app/src/main/java/org/openobservatory/ooniprobe/test/test/RiseupVPN.java

```diff
@@ -27,8 +29,18 @@ public RiseupVPN() {
 
     @Override public void onEntry(Context c, PreferenceManager pm, @NonNull JsonResult json, Measurement measurement) {
         super.onEntry(c, pm, json, measurement);
-        boolean isTransportBlocked = json.test_keys.transport_status.getOrDefault("openvpn", "ok").equals("blocked") ||
-                json.test_keys.transport_status.getOrDefault("obfs4", "ok").equals("blocked");
+        boolean isTransportBlocked = false;
+            isTransportBlocked = getOrDefault(json.test_keys.transport_status, "openvpn", "ok").equals("blocked") ||
+                    getOrDefault(json.test_keys.transport_status, "obfs4", "ok").equals("blocked");
         measurement.is_anomaly = !json.test_keys.ca_cert_status || json.test_keys.api_failure != null || isTransportBlocked;
     }
+
+    public Object getOrDefault(HashMap map, Object key, Object defaultValue) {
+        //ask your app running more modern API as level 24 (Build.VERSION_CODES.N(ougat))
+        return (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N) ?
+                map.getOrDefault(key, defaultValue) :
+        // if not, then need to solve with similar code of original code in next below
+        ((map.get(key) != null) ? map.get(key) : defaultValue);
```

Comment:
I would suggest:
* Using this function independently from the version of Android so we can ensure the code is being tested and run as often as possible
* Moving it into a separate utility related module
* Changing the name to prevent clash with the builtin function (maybe `getOrDefaultCompat`?)

Reply:
So you would remove the if and always use the last part of the function?

Reply:
Yes that's what I was suggesting.